tempesta_fw/cache.c

```diff
@@ -1160,6 +1160,8 @@ cache_req_process_node(TfwHttpReq *req, unsigned long key,
 	TFW_INC_STAT_BH(cache.hits);
 
 	resp = tfw_cache_build_resp(ce);
+	 if ( tfw_cache_entry_is_live(req, ce) > ce->lifetime)
```

Comment:
This function is called just a few lines earlier. Please use it. There's not need to call it again.

Also, the conditions described in RFC7234 4.2.4. are not covered. One of those conditions is not applicable yet (related to revalidate directive as we don't process it yet), but the one related to `Age` header field probably should be taken into acccount.

Also, what about responses that are not generated from cache? The `Warning` header should be added to those as well.